added new Profile option to enable the sensor display of the thousands separator

5kft opened the pull request:
These changes add a new Profile option that enables the display of the "thousands" separator in sensor values.

Here is an example of the updated Profiles page, showing the new option:

<img width="477" height="872" alt="profile-option" src="https://github.com/user-attachments/assets/8afb432c-a93d-4d54-b357-9b032d2fa336" />

\
When you enable this option, the displayed sensor values will include the thousands separator, e.g.,

<img width="463" height="864" alt="profile-option-enabled" src="https://github.com/user-attachments/assets/7a2d739e-d753-4b1c-85a1-ff65e09f4eed" />

\
Here is an example simple sensor panel with the new Profile option disabled (i.e., currently how InfoPanel displays):

<img width="400" height="93" alt="disabled-example" src="https://github.com/user-attachments/assets/6bcebff0-9f5a-4d6a-b2d2-f488e4ecab78" />

\
Here is the same sensor panel with the new Profile option enabled:

<img width="400" height="93" alt="enabled-example" src="https://github.com/user-attachments/assets/eb1e4b24-fc33-4de9-b857-efcd29854ab6" />

\
I've found this very useful in some of my panels, and I thought that others might as well, so I figured I'd try to push it upstream 👍

Thanks for considering this change!

5kft committed:
added new profile option to enable the sensor display of the thousands separator

habibrehmansg commented:
This is a neat addition, however i'm wondering that it should be at the sensor level instead, although that would add some fatigue to update all the items individually. What do you think?

5kft commented:
Indeed...I was actually wondering the same thing when I did this for myself (!).  I ended up thinking that people would probably prefer consistency for something like this - i.e., if they like separators for values in one place in a panel, they'd probably like them the same everywhere in the panel.  And setting this at the panel level one can create different panels that use a different setting for this.  However, I could definitely see that it would make sense at the sensor level - but it would definitely be tedious (but manageable).  So.....excellent question (lol!)

fweepa committed:
Update README.md

habibrehmansg committed:
Update LibreHardwareMonitor and system packages

Bump LibreHardwareMonitorLib from 0.9.5-pre572 to 0.9.6 and update related system packages. System.Diagnostics.PerformanceCounter and System.IO.Ports upgraded to 10.0.3. packages.lock.json regenerated to reflect updated resolutions and transitive bumps (e.g. DiskInfoToolkit 1.1.2, BlackSharp.Core 1.0.7, System.Management 10.0.2, System.CodeDom 10.0.2, ConfigurationManager/EventLog -> 10.0.3, and runtime native System.IO.Ports assets). This brings dependency fixes/clarity into the project lockfile.

habibrehmansg committed:
Update PawnIO_setup.exe

habibrehmansg committed:
Update CLAUDE.md

habibrehmansg committed:
Handle TaskbarCreated to re-register tray icon

Register and handle the TaskbarCreated Windows message so the app can re-register its tray icon when Explorer (taskbar) is restarted. Adds user32.dll imports (RegisterWindowMessage, ChangeWindowMessageFilterEx), a MSGFLT_ALLOW constant, HwndSource hook (WndProc), and uses ChangeWindowMessageFilterEx to allow the message when running elevated. Implements Serilog logger for reporting, calls TrayIcon.Unregister/Register on TaskbarCreated, and removes the hook on window closing. Also updates using directives and introduces fields to store the registered message ID and HwndSource reference.

habibrehmansg committed:
Merge pull request #95 from habibrehmansg/bugfix/1.4.x/fix-disappearing-tray-icon

Handle TaskbarCreated to re-register tray icon

habibrehmansg committed:
Merge pull request #96 from habibrehmansg/86-infopanel-131-constantly-accessing-hdd

Add storage monitoring settings and UI

habibrehmansg committed:
Add storage monitoring settings and UI

Introduce storage/disk monitoring controls and plumbing: add Settings properties (LibreHardwareMonitorStorage: bool, LibreHardwareMonitorStorageInterval: int, default true/30s). Wire them into ConfigModel to copy loaded values and restart LibreMonitor when changed. Update LibreMonitor to respect the storage enable flag and set StorageDevice.ThrottleInterval from the configured interval. Add XAML controls on the Settings page for toggling storage monitoring and configuring the polling interval so users can reduce HDD activity by increasing the interval.

habibrehmansg committed:
Merge pull request #97 from habibrehmansg/bugfix/1.4.x/fix-gif-caching-vram-explosion

Evict GPU texture frames for animated images

habibrehmansg committed:
Evict GPU texture frames for animated images

When decoding a frame into a GPU texture, invalidate other cached SKBitmap frames for animated images to prevent VRAM accumulation. Adds a check (Frames > 1 && bitmapFrame.Image != null) and loops through SKBitmapCache to call Invalidate() on all frames except the current one, freeing GPU resources for non-displayed frames.

habibrehmansg committed:
Merge pull request #91 from fweepa/1.3.x

Update README.md with YouTubeLive Plugin link

habibrehmansg committed:
Debounce render invalidation and use BeginInvoke

Add a volatile _renderInvalidationPending flag to avoid redundant/in-flight render invalidations from the timer. Replace Dispatcher.Invoke(..., Priority.Input) with _dispatcher.BeginInvoke(..., DispatcherPriority.Render) so invalidation is scheduled non-blocking at Render priority. Clear the pending flag at the start of PaintSurface. This reduces reentrancy and UI thread blocking from frequent timer callbacks.

habibrehmansg committed:
Merge branch '1.4.x' of https://github.com/habibrehmansg/infopanel into 1.4.x

5kft committed:
Merge branch 'habibrehmansg:1.4.x' into thousands-separator

habibrehmansg commented:
> Indeed...I was actually wondering the same thing when I did this for myself (!). I ended up thinking that people would probably prefer consistency for something like this - i.e., if they like separators for values in one place in a panel, they'd probably like them the same everywhere in the panel. And setting this at the panel level one can create different panels that use a different setting for this. However, I could definitely see that it would make sense at the sensor level - but it would definitely be tedious (but manageable). So.....excellent question (lol!)

I've merged https://github.com/habibrehmansg/infopanel/pull/115 which was inspired by you. Closing this - thanks!

5kft commented:
Thanks @emaspa and @habibrehmansg !
